3.preprocess-features/3a.merge-features/merge-utils.py

```diff
@@ -0,0 +1,268 @@
+
+    Returns
+    -------
+    pandas.DataFrame
+        single-cell data ready for merging with DP output
+    """
+
+    # set path to plate that we want to load
+    cp_plate_path = pathlib.Path(f"{cp_output_path}/Nuclei.csv")
+
+    # determine which columns to load from CP output csv
+    cp_plate_cell_1 = pd.read_csv(cp_plate_path, nrows=1)
+    all_cols = cp_plate_cell_1.columns.to_list()
+    cols_to_load = [
+        "Metadata_Field",
+        "Metadata_Well",
+        "Location_Center_X",
+        "Location_Center_Y",
+    ]
+    # We only want to get CP data from the feature modules below (_ ensures it is found as module name)
+    cp_feature_modules = [
+        "AreaShape_",
+        "Granularity_",
+        "Intensity_",
+        "Neighbors_",
+        "RadialDistribution_",
+        "Texture_",
+    ]
+    # remove CP columns that dont have a feature module as a substring
+    for col in all_cols:
+        if any(col.startswith(feature_module) for feature_module in cp_feature_modules):
+            cols_to_load.append(col)
+
+    # load single-cell plate data using desired columns
+    cp_plate = pd.read_csv(cp_plate_path, usecols=cols_to_load, low_memory=True)
+
+    # add plate metadata
+    cp_plate["Metadata_Plate"] = plate
+
+    # convert well and field to one usable for merging
+    cp_plate["Metadata_Well"] = cp_plate["Metadata_Well"].apply(format_cp_well)
+    cp_plate["Metadata_Field"] = cp_plate["Metadata_Field"].apply(format_cp_site)
+    # rename field column to site (name used by DP)
+    cp_plate = cp_plate.rename(columns={"Metadata_Field": "Metadata_Site"})
+
+    return cp_plate
+
+
+def full_loc_map(dp_coord: tuple, cp_image_data_locations: pd.Series) -> tuple:
+    """
+    helper function for merge_CP_DP_batch_data
+    get cp_coord from cp_image_data_locations that is closest to dp_coord
+
+    Parameters
+    ----------
+    dp_coord : tuple
+        dp coord to find closest cp coord for
+    cp_image_data_locations : pd.Series
+        series of cp coords to get closest one from
+
+    Returns
+    -------
+    tuple
+        closest cp_coord to given dp_coord
+    """
+    return min(
+        cp_image_data_locations,
+        key=lambda cp_coord: math.hypot(
+            cp_coord[0] - dp_coord[0], cp_coord[1] - dp_coord[1]
+        ),
+    )
+
+
+def merge_CP_DP_image_data(
+    cp_image_data: pd.DataFrame,
+    dp_image_data: pd.DataFrame,
+    add_cell_uuid: bool = False,
+) -> pd.DataFrame:
+    """
+    merge CP and DP single-cell data from the same image (plate, well, site combination)
+
+    Parameters
+    ----------
+    cp_image_data : pd.DataFrame
+        cp single-cell data from the image to merge cells from
+    dp_image_data : pd.DataFrame
+        dp single-cell data from the image to merge cells from
+    add_cell_uuid : bool, optional
+        whether or not single-cell UUIDs should be added (useful for identifying a particular cell), by default True
```

Comment:
```suggestion
        whether or not single-cell UUIDs should be added (useful for identifying a particular cell), by default False
```
Consider revising docstring. The `add_cell_uuid` is set to `False` by default in the function parameters.